src/extensions/entity.extension.ts

```diff
@@ -357,16 +366,112 @@ export function EntityManager({
     await refresh();
   });
 
+  // #region Registry
+  async function AddLabel({ entity, label }: EditLabelOptions) {
+    const current = await EntityGet(entity);
+    if (current?.labels?.includes(label)) {
+      logger.debug({ name: entity }, `already has label {%s}`, label);
+      return;
+    }
+    await hass.socket.sendMessage({
+      entity_id: entity,
+      labels: [...current.labels, label],
+      type: UPDATE_REGISTRY,
+    });
+  }
+
+  async function EntitySource() {
+    return await hass.socket.sendMessage<
+      Record<PICK_ENTITY, { domain: string }>
+    >({ type: "entity/source" });
+  }
+
+  async function EntityList() {
+    return await hass.socket.sendMessage<EntityDetails[]>({
+      type: "config/entity_registry/list_for_display",
+    });
+  }
+
+  async function RemoveLabel({ entity, label }: EditLabelOptions) {
+    const current = await EntityGet(entity);
+    if (!current?.labels?.includes(label)) {
+      logger.debug({ name: entity }, `does not have label {%s}`, label);
+      return;
+    }
+    logger.debug({ name: entity }, `removing label [%s]`, label);
+    await hass.socket.sendMessage({
+      entity_id: entity,
+      labels: current.labels.filter(i => i !== label),
+      type: UPDATE_REGISTRY,
+    });
+  }
+
+  async function EntityGet<ENTITY extends PICK_ENTITY>(entity_id: ENTITY) {
+    return await hass.socket.sendMessage<EntityRegistryItem<ENTITY>>({
+      entity_id: entity_id,
+      type: "config/entity_registry/get",
+    });
+  }
+
+  lifecycle.onBootstrap(async () => {
+    if (!config.hass.AUTO_CONNECT_SOCKET || !config.hass.MANAGE_REGISTRY) {
+      return;
+    }
+    hass.entity.registry.current = await hass.entity.registry.list();
+    hass.socket.subscribe({
+      context,
+      event_type: "entity_registry_updated",
+      async exec() {
+        logger.debug("entity registry updated");
+        hass.entity.registry.current = await hass.entity.registry.list();
+      },
+    });
+  });
+
+  // #endregion
+  function byLabel(label: TLabelId): PICK_ENTITY[] {
+    return hass.entity.registry.current
+      .filter(i => i.labels.includes(label))
+      .map(i => i.entity_id);
```

Comment:
The FP programmer in me itches at double iterations, but this change is largely irrelevant, just for your consideration

```suggestion
    return hass.entity.registry.current
      .reduce(((acc, i) => i.labels.includes(label) ? [...acc, i.entity_id] : acc), []);
```